### ISSUE TITLE
Challenger's Peril affix not tracked properly since its update

### ISSUE THREAD

**Issue by Resxt** [ipzaur/ipmythictimer#17]
Hello,

Recently the [Challenger's Peril affix got a change that adds 90 seconds to the timer](https://www.wowhead.com/news/challengers-peril-affix-nerfed-on-weekly-reset-bonus-dungeon-timer-when-affix-349282)

I recently did a Dawnbreaker with something like 06:12 left on timer which is below the 06:30 that's the very minimum to have it be a +2 push yet the key was still pushed +2 (WOW chat and raider.io as proof)
To add to this, I can confirm that the IPMythicTimer was showing that the key will be a +1 push close to the timer before we killed the last boss

TLDR: Challenger's Peril affix gives 90 more seconds to the timer since a few days but the addon doesn't take this change into account for the +1/+2/+3 push

I hope that you can update the addon to fix this issue, thanks 👍 
Great addon by the way

**Comment by ipzaur** [comment]
@Resxt Hi!
Thanks for report and warm words. I'll try to fix it.

**Comment by ipzaur** [comment]
@Resxt 
Hello again!
Do you have any screenshots or videos to help me with analysing this problem?
I have analysed other addons for the M+ timer, and they all count the time for +2 and +3 considering 90 seconds, just like my addon. For example, if we take Dawnbreaker, it has base time of 31 minutes. With the affix, all addons for timer (and mine too) show 32:30.
The formula for +3 and +2 is calculated as ([base time] * 0.6 or 0.8).
Accordingly, in our case, for +3 we need to run the dungeon in 19:30 (or 13 minutes left on the timer). For +2 we need to run the dungeon in 26:00 (6:30 on the timer).
I tried to calculate the timers for +2 and +3 without the 90 seconds on top. And it seems to work out the way we need it to:
For +3 we need to complete run in 18:36 (12:24 on the timer), and for +2 we need to run in 24:48 (at least 6:12 on the timer).
But I'm not sure yet if it's really necessary to count these cutoffs without counting 90 seconds (at this moment other addons don't do this, and mine too). If you have something to help me analyse this, I'd be glad for help :)

**Comment by Resxt** [comment]
Hello

No sadly I don't have any media to share for this, I simply remember telling my friend "so close to getting it to +2" (it was like 5-10 seconds below +2 timer) and he told me "well it is, just look at the chat" and the timer addon was showing +1

I can't provide anything more for now, I'll try to stay on the look for potential +2 going to +1 and screenshot it if it happens again 

Thanks for looking into it, hopefully I'm not wasting your time

**Comment by ipzaur** [comment]
@Resxt 
Whohoo!
I think I'm understand how to properly calculate cutoffs.
Just need to calculate the time for +2 and +3 for the base timer (without adding 90sec) and add 90sec to these cutoffs.
At least for Dawnbreaker and GrimBathol everything fits (I looked at +2 runs on raider.io to analyse).
For example, let's take Dawnbreaker:
Base timer is 31:00, and with an affix it is 32:30.
The old calculation was: 32:30 * 0.8 = 26:00 (6:30 left on the timer), which is not correct, as we found out.
The correct count is: 31:00 * 0.8 = 24:48 + 90 seconds = 26:18 (6:12 left on the timer).
I have upload a new build of addon with a new formula for the "Challenger's Peril" affix. Unfortunately I'm not playing the game yet, but I'm trying to maintain the addon. In theory, the timing for +2 and +3 should now work correctly. Please try it and let me know the result ;)

**Comment by Resxt** [comment]
It's gonna take quite some time to have specifically a key that's close to depleting to a +1 push but I'll try to keep an eye on it and provide feedback whenever I can

Thank you!

**Comment by Resxt** [comment]
Just did a Dawnbreaker, timer wasn't close enough to have the "is it +2 or +1?" issue when I took the screenshot since it was in the middle of last boss but I took this screenshot (before your update) in case it can be of any use 🤷 

![image](https://github.com/user-attachments/assets/9326c8b0-df1c-40f2-83a4-7fb776fcddca)

**Comment by Resxt** [comment]
After testing for several days I had several encounters where the key was almost depleted or close to going down to a +1 push and I didn't notice any issue, everything seems to work normally now 👍

Thank you once again
